nodebuilder/tests/tastora/blob_test.go

```diff
@@ -31,319 +34,432 @@ func TestBlobTestSuite(t *testing.T) {
+				s.Assert().Equal(data1, retrievedData1, "%s namespace1 data should match", nodeType)
+			}
+
+			// Get all blobs for namespace2 - should only return namespace2 data
+			allBlobs2, err := client.Blob.GetAll(ctx, height2, []share.Namespace{namespace2})
+			s.Require().NoError(err, "%s should get namespace2 blobs", nodeType)
+			s.Assert().Len(allBlobs2, 1, "%s should find exactly 1 blob in namespace2", nodeType)
+
+			if len(allBlobs2) > 0 {
+				retrievedData2 := bytes.TrimRight(allBlobs2[0].Data(), "\x00")
+				s.Assert().Equal(data2, retrievedData2, "%s namespace2 data should match", nodeType)
+			}
+
+			// Cross-namespace verification - namespace1 query should not return namespace2 data
+			crossBlobs, err := client.Blob.GetAll(ctx, height2, []share.Namespace{namespace1})
+			s.Require().NoError(err)
+			// Should be empty since namespace1 blob was submitted at height1, not height2
+			s.Assert().Empty(crossBlobs, "%s should not find namespace1 blobs at height2", nodeType)
+
+			s.T().Logf("✅ %s namespace isolation verified", nodeType)
+		})
 	}
 }
 
-// TestBlobGet_ExistingBlob tests blob retrieval API for existing blobs
-func (s *BlobTestSuite) TestBlobGet_ExistingBlob() {
-	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+// TestConcurrentBlobOperations validates that multiple concurrent blob operations
+// work correctly across different nodes without conflicts
+func (s *BlobTestSuite) TestConcurrentBlobOperations() {
+	ctx, cancel := context.WithTimeout(context.Background(), 6*time.Minute)
 	defer cancel()
 
 	bridgeNode := s.framework.GetBridgeNodes()[0]
-	client := s.framework.GetNodeRPCClient(ctx, bridgeNode)
-
-	namespace, err := share.NewV0Namespace(bytes.Repeat([]byte{0x03}, 10))
-	s.Require().NoError(err)
-
-	nodeAddr, err := client.State.AccountAddress(ctx)
-	s.Require().NoError(err)
+	lightNode1 := s.framework.NewLightNode(ctx)
+	lightNode2 := s.framework.NewLightNode(ctx)
+
+	bridgeClient := s.framework.GetNodeRPCClient(ctx, bridgeNode)
+	light1Client := s.framework.GetNodeRPCClient(ctx, lightNode1)
+	light2Client := s.framework.GetNodeRPCClient(ctx, lightNode2)
+
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	results := make(map[string]uint64) // Map submission ID to block height
+
+	// Concurrent blob submissions
+	numSubmissions := 3
+	for i := 0; i < numSubmissions; i++ {
+		wg.Add(1)
+		go func(submissionID int) {
+			defer wg.Done()
+
+			// Create unique namespace and data for this submission
+			namespaceBytes := bytes.Repeat([]byte{byte(0x30 + submissionID)}, 10)
+			namespace, err := share.NewV0Namespace(namespaceBytes)
+			if err != nil {
+				s.T().Errorf("Submission %d: failed to create namespace: %v", submissionID, err)
+				return
+			}
+
+			data := []byte(fmt.Sprintf("Concurrent submission %d data", submissionID))
+			nodeBlobs := s.createBlobsForSubmission(ctx, bridgeClient, namespace, data)
+
+			// Submit with different gas to avoid nonce conflicts
```

Comment:
The `txClient` should be able to handle nonce conflicts internally at some point. I'd prefer if we had a test case either now (which we can expect to flake) or later on once we have confirmed better nonce mgmt for concurrent txs that will submit txs with the defaults and ensure they are committed if valid.